### PR TITLE
chore: switch implementation addresses in README table to M2 versions

### PR DESCRIPTION
**Pull request by ChaoticWalrus**
Marking this as a draft until upgrade is actually performed, since the implementations have not actually changed yet.

### CODE REVIEW

**Comment by wadealexc** [comment]
Should update the "preview" of each address as well. Each link is accompanied with a slice of 4 bytes of the address, e.g. `0xAABB...CCDD`

**Comment by ChaoticWalrus** [comment]
> Should update the "preview" of each address as well. Each link is accompanied with a slice of 4 bytes of the address, e.g. `0xAABB...CCDD`

ahh, my bad, good catch -- these should be fixed now.

File: README.md [hunk 1]
```diff
@@ -107,13 +107,13 @@ and/or
 
 | Name | Solidity | Proxy | Implementation | Notes |
 | -------- | -------- | -------- | -------- | -------- | 
-| StrategyManager | [`StrategyManager`](https://github.com/Layr-Labs/eigenlayer-contracts/blob/0139d6213927c0a7812578899ddd3dda58051928/src/contracts/core/StrategyManager.sol) | [`0x779...8E907`](https://goerli.etherscan.io/address/0x779d1b5315df083e3F9E94cB495983500bA8E907) | [`0x3FFa...3dE6`](https://goerli.etherscan.io/address/0x3FFa9daE46d15f15c925d4694c19c49e624b3dE6) | Proxy: [OpenZeppelin TUP@4.7.1](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.7.1/contracts/proxy/transparent/TransparentUpgradeableProxy.sol) |
+| StrategyManager | [`StrategyManager`](https://github.com/Layr-Labs/eigenlayer-contracts/blob/0139d6213927c0a7812578899ddd3dda58051928/src/contracts/core/StrategyManager.sol) | [`0x779...8E907`](https://goerli.etherscan.io/address/0x779d1b5315df083e3F9E94cB495983500bA8E907) | [`0x8676...0055`](https://goerli.etherscan.io/address/0x8676bb5f792ED407a237234Fe422aC6ed3540055) | Proxy: [OpenZeppelin TUP@4.7.1](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.7.1/contracts/proxy/transparent/TransparentUpgradeableProxy.sol) |
```
Comment (wadealexc) [inline README.md:110]: Just noticing that a handful of these addresses use really inconsistent "preview" formats...

Fix now or in a later PR?

i.e. the mainnet contract addresses above all preview 2 leading bytes and 2 ending bytes.

These goerli addresses just sorta... do a random amount of bytes.

**Review by wadealexc — APPROVED** [review]
Left some notes above, but LGTM generally - we can follow up with additional changes, or tack onto this PR - your choice.

Two more things:
1. I wonder if we should make a folder that sorta copies over these old files, and just keep the current stuff in the README. So instead of just deleting the old impl addresses here, we copy the table as it looks now and add it to `deployments/m1.md` or something like that.
2. When you merge, please squash and merge! :heart:

File: README.md (hunk 1 above)
Comment (wadealexc) [reply README.md:110]: Later PR is fine, no worries. I'll make an issue

Comment (wadealexc) [reply README.md:110]: https://github.com/Layr-Labs/eigenlayer-contracts/issues/331

**Comment by ChaoticWalrus** [comment]
Good suggestions, going to (squash) merge this as-is for the moment.

**Pull request merged**